sound/soc/sof/ipc3-topology.c

```diff
@@ -2349,10 +2349,7 @@ static int sof_ipc3_parse_manifest(struct snd_soc_component *scomp, int index,
 	}
 
 	if (SOF_ABI_VERSION_MINOR(abi_version) > SOF_ABI_MINOR) {
-		if (!IS_ENABLED(CONFIG_SND_SOC_SOF_STRICT_ABI_CHECKS)) {
-			dev_warn(scomp->dev, "%s: Topology ABI is more recent than kernel\n",
-				 __func__);
-		} else {
+		if (IS_ENABLED(CONFIG_SND_SOC_SOF_STRICT_ABI_CHECKS)) {
```

Comment:
condense it to a single `if` and check only the ABI version _if_ the strict ABI check is enabled only?
```c
	if (IS_ENABLED(CONFIG_SND_SOC_SOF_STRICT_ABI_CHECKS) &&
	    (SOF_ABI_VERSION_MINOR(abi_version) > SOF_ABI_MINOR)) {
		dev_err(scomp->dev, "%s: Topology ABI is more recent than kernel\n",
			__func__);
		return -EINVAL;
	}
```

Reply:
done